### PR TITLE
build: Patch socket.io packages for node16 moduleResolution

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR patches socket.io-client and engine.io-client, applying the changes from https://github.com/socketio/socket.io-client/pull/1595 and https://github.com/socketio/engine.io-client/pull/711, respectively.

I think this is a safe change because we are using only types from the package, and the .d.ts/exported API still refers to the real socket.io-client package, so consumers should see no difference - they might still hit the same bug we're hitting, but only when using node16 moduleResolution.

[AB#6140](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/6140)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -349,7 +349,9 @@
 			"qs": "^6.11.0",
 			"simplemde>codemirror": "^5.65.11",
 			"simplemde>marked": "^4.3.0",
-			"@oclif/core": "~2.4.0"
+			"@oclif/core": "~2.4.0",
+			"@fluidframework/eslint-config-fluid": "~3.1.0",
+			"eslint-plugin-import": "npm:eslint-plugin-i@~2.29.0"
```
Comment (tylerbutler) [inline package.json:354]: These can be removed once #18168 is merged.

File: syncpack.config.cjs
Comment (tylerbutler) [inline syncpack.config.cjs:1]: Changes in this file can be reverted once #18168 is merged.

**Comment by Josmithr** [comment]
Could we add a README in the `patches` directory (assuming we don't already have one) and add docs explaining what the patches are, why we need them, and link to the appropriate work items whose completion would enable us to remove these?

**Comment by tylerbutler** [comment]
> Could we add a README in the `patches` directory (assuming we don't already have one) and add docs explaining what the patches are, why we need them, and link to the appropriate work items whose completion would enable us to remove these?

Added a readme.

**Review by Josmithr — APPROVED** [review]
Thanks for adding the docs!

**Pull request merged**